code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -681,11 +696,13 @@ GLOBAL_VAR_INIT(dynamic_forced_threat_level, -1)
 	return type_list
 
 /// Checks if a type in blocking_list is in rule_list.
-/datum/game_mode/dynamic/proc/check_blocking(list/blocking_list, list/rule_list)
+/datum/game_mode/dynamic/proc/check_blocking(list/blocking_list, list/rule_list, ignore_dead_rulesets = TRUE)
 	if(blocking_list.len > 0)
 		for(var/blocking in blocking_list)
 			for(var/_executed in rule_list)
-				var/datum/executed = _executed
+				var/datum/dynamic_ruleset/executed = _executed
+				if((ignore_dead_rulesets && !high_impact_major_event_occured) && executed.is_dead())
```

Comment:
optimization: skip the is_dead check if it's not actually a blocking type. so move all this into the if below. best way is converting the if to its inverse, adding a continue, then putting all the logic in a block after the new statement

```
if(blocking != executed.type)
    continue
if((ignore_dead ... )
    continue
log_game("DYNAMIC: FAIL: ")
return TRUE
```

Reply:
Done